src/tests/ftest/control/dmg_nvme_scan_test.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2018-2019 Intel Corporation.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+kworker/16:1
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+import time
+
+from dmg_utils import DmgCommand
+from server_utils import ServerCommand
+from apricot import TestWithoutServers
+from avocado.utils import process
+
+class DmgNvmeScanTest(TestWithoutServers):
+    """ Simple test to verify the scan function of the dmg tool.
+    :avocado: recursive
+    """
+    def __init__(self, *args, **kwargs):
+        super(DmgNvmeScanTest, self).__init__(*args, **kwargs)
+
+    def clean_up(self):
+        """Setup/cleanup for the daos_server to run properly."""
+        cmd_list = [
+            "umount /mnt/daos; rm -rf /mnt/daos",
+            "rm -rf /tmp/daos_sockets/",
+            "rm -rf /tmp/*.log",
+        ]
+        for cmd in cmd_list:
+            try:
+                process.run(cmd, verbose=True, sudo=True)
+            except process.CmdError as error:
+                self.fail("Error issuing '{}': {}".format(cmd, error))
+
+    def test_dmg_nvme_scan_basic(self):
+        """ Test basic dmg functionality to scan nvme the storage on system.
+        :avocado: tags=all,hw,dmg,control
+        """
+        self.clean_up()
+
+        # Create daos_server command
+        server = ServerCommand()
+        server.get_params(self, "/run/daos_server/*")
+
+        # Edit the config value to full path
+        server.config.value = os.path.join(
+            self.basepath, str(server.config.value))
+
+        # Run server as a background process.
+        svr_subproc = server.subproc()
+        try:
+            pid = svr_subproc.start()
+        except process.CmdError as details:
+            self.fail("Server command failed: {}".format(details))
+
+        # Wait for server to start
+        time.sleep(3)
```

Comment:
We should do something more deterministic.   The current server_utils.py run_server() function uses the detection of the "DAOS I/O Server" string in the subprocess output to determine that the server has started.  I would suggest including this check in a `ServerCommand.start()` method and including support for using orterun to run the server on a differnt host than the host launching the test:

```
    def get_launch_command(self, manager, uri=None, env=None):
        """Get the process launch command used to run daos_server.

        Args:
            manager (str): mpi job manager command
            uri (str, optional): path to uri file. Defaults to None.
            env (dict, optional): dictionary on env variable names and values.
                Defaults to None.

        Raises:
            ServerFailed: if the specified job manager is unsupported.

        Returns:
            str: daos_sever launch command

        """
        if manager is None:
            # Run locally
            return self.__str__()

        elif manager.endswith("orterun"):
            assign_env = ["{}={}".format(key, val) for key, val in env.items()]
            args = [
                "-np {}".format(len(self.hosts)),
                "-hostfile {}".format(self.hostfile),
                "--enable-recovery",
            ]
            args.extend(["-x {}".format(assign) for assign in assign_env])
            if uri is not None:
                args.append("--report-uri {}".format(uri))

        else:
            raise ServerFailed("Unsupported job manager: {}".format(manager))

        return "{} {} {}".format(manager, " ".join(args), self.__str__())

    def start(self, manager, verbose=True, env=None, timeout=600):
        """Start the daos server on each specified host.

        Args:
            manager (str): mpi job manager command
            verbose (bool, optional): [description]. Defaults to True.
            env ([type], optional): [description]. Defaults to None.
            timeout (int, optional): [description]. Defaults to 600.

        Raises:
            ServerFailed: if there are issues starting the servers

        """
        if self.process is None:
            # Start the daos server as a subprocess
            kwargs = {
                "cmd": self.get_launch_command(manager),
                "verbose": verbose,
                "allow_output_check": "combined",
                "shell": True,
                "env": env,
                "sudo": True,
            }
            self.process = process.SubProcess(**kwargs)
            self.process.start()

            # Wait for 'DAOS I/O server' messages to appear in the daos_server
            # output indicating that the servers have started
            start_time = time.time()
            start_msgs = 0
            timed_out = False
            while start_msgs != len(self.hosts) and not timed_out:
                output = self.process.get_stdout()
                start_msgs = len(re.findall("DAOS I/O server", output))
                timed_out = time.time() - start_time > timeout

            if start_msgs != len(self.hosts):
                err_msg = "{} detected starting {}/{} servers".format(
                    "Timed out" if timed_out else "Error",
                    start_msgs, len(self.hosts))
                print("{}:\n{}".format(err_msg, self.process.get_stdout()))
                raise ServerFailed(err_msg)
```

Reply:
Will make this change.